src/System.Diagnostics.Process/src/System/Diagnostics/Process.Win32.cs

```diff
@@ -0,0 +1,168 @@
+            fixed (char* verb = startInfo.Verb.Length > 0 ? startInfo.Verb : null)
+            fixed (char* parameters = startInfo.Arguments.Length > 0 ? startInfo.Arguments : null)
+            fixed (char* directory = startInfo.WorkingDirectory.Length > 0 ? startInfo.WorkingDirectory : null)
+            {
+                Interop.Shell32.SHELLEXECUTEINFO shellExecuteInfo = new Interop.Shell32.SHELLEXECUTEINFO()
+                {
+                    cbSize = (uint)sizeof(Interop.Shell32.SHELLEXECUTEINFO),
+                    lpFile = fileName,
+                    lpVerb = verb,
+                    lpParameters = parameters,
+                    lpDirectory = directory,
+                    fMask = Interop.Shell32.SEE_MASK_NOCLOSEPROCESS | Interop.Shell32.SEE_MASK_FLAG_DDEWAIT
+                };
+
+                if (startInfo.ErrorDialog)
+                    shellExecuteInfo.hwnd = startInfo.ErrorDialogParentHandle;
+                else
+                    shellExecuteInfo.fMask |= Interop.Shell32.SEE_MASK_FLAG_NO_UI;
+
+                switch (startInfo.WindowStyle)
+                {
+                    case ProcessWindowStyle.Hidden:
+                        shellExecuteInfo.nShow = Interop.Shell32.SW_HIDE;
+                        break;
+                    case ProcessWindowStyle.Minimized:
+                        shellExecuteInfo.nShow = Interop.Shell32.SW_SHOWMINIMIZED;
+                        break;
+                    case ProcessWindowStyle.Maximized:
+                        shellExecuteInfo.nShow = Interop.Shell32.SW_SHOWMAXIMIZED;
+                        break;
+                    default:
+                        shellExecuteInfo.nShow = Interop.Shell32.SW_SHOWNORMAL;
+                        break;
+                }
+
+                ShellExecuteHelper executeHelper = new ShellExecuteHelper(&shellExecuteInfo);
+                if (!executeHelper.ShellExecuteOnSTAThread())
+                {
+                    int error = executeHelper.ErrorCode;
+                    if (error == 0)
+                    {
+                        error = GetShellError(shellExecuteInfo.hInstApp);
+                    }
+
+                    switch (error)
+                    {
+                        case Interop.Errors.ERROR_BAD_EXE_FORMAT:
+                        case Interop.Errors.ERROR_EXE_MACHINE_TYPE_MISMATCH:
+                            throw new Win32Exception(error, SR.InvalidApplication);
+                        default:
+                            throw new Win32Exception(error);
+                    }
+                }
+
+                if (shellExecuteInfo.hProcess != IntPtr.Zero)
+                {
+                    SetProcessHandle(new SafeProcessHandle(shellExecuteInfo.hProcess));
+                    return true;
+                }
+            }
+
+            return false;
+        }
+
+        private int GetShellError(IntPtr error)
+        {
+            switch ((long)error)
```

Comment:
It could, after casting to long, be cast to int. That would, however, change the matching on 64 bit if for some reason there was something in the high int. Keeping this in line with the NetFX implementation. Note, however, that this code will likely never be hit- there should always be a LastError with ShellExecuteEx() afaik. Not 100% positive there aren't corner cases, however, and want to stay NetFX aligned as a result.